lightning/src/ln/channelmanager.rs

```diff
@@ -3015,22 +3016,18 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 				}}
 			}
 
-			macro_rules! add_update_add_htlc {
-				($add_htlc_msg: expr, $channel_id: expr, $counterparty_node_id: expr) => {{
+			macro_rules! add_update_htlc_msg {
+				($htlc_msg: expr, $channel_id: expr, $counterparty_node_id: expr) => {{
 					add_channel_key!($channel_id, $counterparty_node_id);
 					if let hash_map::Entry::Occupied(mut entry) = htlcs_msgs_by_id.entry($channel_id) {
 						let msgs_entry = entry.get_mut();
-						msgs_entry.0.push($add_htlc_msg);
-					}
-				}}
-			}
-
-			macro_rules! add_update_fail_htlc {
-				($fail_htlc_msg: expr, $channel_id: expr, $counterparty_node_id: expr) => {{
-					add_channel_key!($channel_id, $counterparty_node_id);
-					if let hash_map::Entry::Occupied(mut entry) = htlcs_msgs_by_id.entry($channel_id) {
-						let msgs_entry = entry.get_mut();
-						msgs_entry.1.push($fail_htlc_msg);
+						if let Some(msg) = (&$htlc_msg as &Any).downcast_ref::<msgs::UpdateAddHTLC>() {
```

Comment:
Definitely not happy with this downcast solution, but couldn't get a match statement to work. If you can think of a better way of doing this, or if you prefer avoiding this DRY up, please let me know!